src/HealthGPS.Core/column_primitive.h

```diff
@@ -113,6 +113,9 @@ template <typename TYPE> class PrimitiveDataTableColumn : public DataTableColumn
     /// @return An iterator to the end
     IteratorType end() const { return IteratorType(*this, size()); }
 
+    /// Returns the underlying column data as a vector.
+    std::vector<value_type> values() { return data_; }
```

Comment:
This should return a const ref, otherwise it'll unconditionally copy the vector. If you return a const ref then the caller has the option of copying it, but can also just use it as a reference.

```suggestion
    const std::vector<value_type> &values() { return data_; }
```